rlp/decode.go

```diff
@@ -211,6 +213,22 @@ func decodeUint(s *Stream, val reflect.Value) error {
 	return nil
 }
 
+func decodeBool(s *Stream, val reflect.Value) error {
+	num, err := s.uint(8)
+	if err != nil {
+		return wrapStreamError(err, val.Type())
+	}
+	switch num {
+	case 0:
+		val.SetBool(false)
+	case 1:
+		val.SetBool(true)
+	default:
+		return fmt.Errorf("rlp: invalid boolean value: %d", num)
+	}
+	return nil
+}
```

Comment:
The advantage will be nicer error messages (with struct field context) through `wrapStreamError`.